src/nypl/mod.rs

```diff
@@ -0,0 +1,204 @@
+use crate::Vec2d;
+use std::str::FromStr;
+use custom_error::custom_error;
+use std::sync::Arc;
+use crate::dezoomer::{TilesRect, Dezoomer, DezoomerInput, ZoomLevels, DezoomerError, IntoZoomLevels, DezoomerInputWithContents};
+use std::convert::TryFrom;
+use std::iter::successors;
+use std::fmt::Debug;
+use serde::export::Formatter;
+
+/// A dezoomer for NYPL images
+#[derive(Default)]
+pub struct NYPLImage;
+
+const NYPL_IMAGE_VIEW_PREFIX: &str = "https://digitalcollections.nypl.org/items/";
+const NYPL_META_PREFIX: &str = "https://access.nypl.org/image.php/";
+const NYPL_META_POSTFIX: &str = "/tiles/config.js";
+
+fn _get_image_url_from_meta_url(meta_url: &str) -> String {
```

Comment:
If this function is used, its name shouldn't start with an underscore

Reply:
Will be done.